pasta_eln/GUI/form.py

```diff
@@ -344,107 +344,111 @@ def execute(self, command:list[Any]) -> None:
-        if key in ['comment','content']:
-          text = getattr(self, f'textEdit_{key}').toPlainText().strip()
-          if '_ids' not in self.doc or text:  #if group edit, text has to have text
-            self.doc[key] = text
-            if key == 'content' and '-branch' in self.doc:
-              for branch in self.doc['-branch']:
-                if branch['path'] is not None:
-                  if branch['path'].endswith('.md'):
-                    with open(self.comm.backend.basePath/branch['path'], 'w', encoding='utf-8') as fOut:
-                      fOut.write(self.doc['content'])
-                    logging.debug('Wrote new content to '+branch['path'])
-                  else:
-                    showMessage(self, 'Information', 'Did update the database but not the file on harddisk, since PASTA-ELN cannot write this format')
-        elif isinstance(valueOld, list):  #items that are comma separated in the text-field
-          self.doc[key] = getattr(self, f'key_{key}').text().strip().split(' ')
-        elif isinstance(valueOld, str):
-          if isinstance(getattr(self, f'key_{key}'), QComboBox):
-            valueNew = getattr(self, f'key_{key}').currentText()
-            if (valueNew != '- no link -'
-                and getattr(self, f'key_{key}').currentData() is not None
-                and re.search(
-                    r"^[a-z\-]-[a-z0-9]{32}$",
-                    getattr(self, f'key_{key}').currentData(),
-                ) is not None):
-              #if docID is stored in currentData
-              self.doc[key] = getattr(self, f'key_{key}').currentData()
-            elif valueNew!='- no link -' :
-              self.doc[key] = valueNew
-          else:                          #normal text field
-            self.doc[key] = getattr(self, f'key_{key}').text().strip()
-        elif valueOld is None and key in self.doc:  #important entry, set to empty string
-          self.doc[key]=''
-        else:
-          print("**ERROR dialogForm unknown value type",key, valueOld)
-      # new key-value pairs
-      keyValueList = [self.keyValueListL.itemAt(i).widget().text() for i in range(self.keyValueListL.count())]
-      keyValueDict = dict(zip(keyValueList[::2],keyValueList[1::2] ))
-      keyValueDict = {k:v for k,v in keyValueDict.items() if k}
-      self.doc = keyValueDict | self.doc
-      # ---- if project changed: only branch save; remaining data still needs saving
-      newProjID = []
-      if hasattr(self, 'projectComboBox') and self.projectComboBox.currentData() != '':
-        parentPath = self.db.getDoc(self.projectComboBox.currentData())['-branch'][0]['path']
-        if '_ids' in self.doc:  # group update
-          for docID in self.doc['_ids']:
-            doc = self.db.getDoc(docID)
-            if doc['-branch'][0]['stack']!=self.projectComboBox.currentData(): #only if project changed
-              if doc['-branch'][0]['path'] is None:
+      from cProfile import Profile
```

Comment:
Do you want to include profiling in production code?

Since it's clearly mentioned in the docs that this will introduce overhead: https://docs.python.org/2/library/profile.html#introduction-to-the-profilers